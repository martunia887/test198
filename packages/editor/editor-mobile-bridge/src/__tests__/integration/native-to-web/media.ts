import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  callNativeBridge,
  editor,
  editable,
  getDocFromElement,
  skipBrowsers as skip,
} from '../_utils';

BrowserTestCase(
  `media.ts: Insert media single having dimensions`,
  { skip },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    const uploadPreviewUpdatePayload = {
      file: {
        dimensions: {
          width: 2265,
          height: 1500,
        },
        id: 'cc6359f2-dcd6-47f9-ae22-4ac8b86cddb2',
        name: 'rose-blue-flower-rose-blooms-67636.jpeg',
        type: 'image/jpeg',
      },
      preview: {
        dimensions: {
          width: 2265,
          height: 1500,
        },
      },
    };

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(uploadPreviewUpdatePayload),
    );

    const mobileUploadEndPayload = {
      file: {
        collectionName: 'TestCollection',
        id: 'cc6359f2-dcd6-47f9-ae22-4ac8b86cddb2',
        publicId: '12d5234f-eb29-424a-84fe-36fe14a33754',
        name: 'rose-blue-flower-rose-blooms-67636.jpeg',
        type: 'image/jpeg',
      },
    };

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-end',
      JSON.stringify(mobileUploadEndPayload),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  `media.ts: Insert media single unknown dimensions`,
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    const uploadPreviewUpdatePayload = {
      file: {
        id: 'dc6359f2-dcd6-47f9-ae22-4ac8b86cddb2',
        name: 'rose-blue-flower-rose-blooms-67636.jpeg',
        type: 'image/jpeg',
      },
    };

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-preview-update',
      JSON.stringify(uploadPreviewUpdatePayload),
    );

    const mobileUploadEndPayload = {
      file: {
        collectionName: 'TestCollection',
        id: 'dc6359f2-dcd6-47f9-ae22-4ac8b86cddb2',
        publicId: 'e2d5234f-eb29-424a-84fe-36fe14a33754',
        name: 'rose-blue-flower-rose-blooms-67636.jpeg',
        type: 'image/jpeg',
      },
    };

    await callNativeBridge(
      browser,
      'onMediaPicked',
      'upload-end',
      JSON.stringify(mobileUploadEndPayload),
    );

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
