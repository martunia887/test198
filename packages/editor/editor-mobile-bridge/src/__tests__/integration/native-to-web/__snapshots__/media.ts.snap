// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media.ts: Insert media single having dimensions 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "TestCollection",
            "height": 1500,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 2265,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`media.ts: Insert media single unknown dimensions 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "layout": "center",
        "width": null,
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__displayType": null,
            "__fileMimeType": "image/jpeg",
            "__fileName": "example.png",
            "__fileSize": null,
            "collection": "TestCollection",
            "height": 200,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": 250,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
